### ISSUE TITLE
Not clear logic for loading CMS Pages with setStoreId function 

### ISSUE THREAD

**Issue by Sharkozp** [magento/magento2#22767]
### Preconditions (*)

1. 2.1.x
2. 2.2.x
3. 2.3.x

### Steps to reproduce (*)

1. Create some class that will load CMS Page:
```
 /**
     * Constructor.
     */
    public function __construct(
        \Magento\Cms\Model\PageFactory $pageFactory
    ) {
        $this->pageFactory = $pageFactory;
    }

/** @var \Magento\Cms\Model\Page $page */
$page = $this->pageFactory->create();
$page->setStoreId([2]);
$page->load($identifier, Page::IDENTIFIER);
```
2. According to description for magic function **setStoreId** must be array in Magento\Cms\Model\Page:
` * @method Page setStoreId(array $storeId)`

[Link](https://github.com/magento/magento2/blob/bd82ed4ecd5c7d66c819711a2a4425d46c232583/app/code/Magento/Cms/Model/Page.php#L19)

3. But in Magento\Cms\Model\ResourceModel\Page in method **_getLoadSelect** you have this part of code:
```
 if ($object->getStoreId()) {
            $storeIds = [
                Store::DEFAULT_STORE_ID,
                (int)$object->getStoreId(),
            ];
......
}
```
[Link](https://github.com/magento/magento2/blob/bd82ed4ecd5c7d66c819711a2a4425d46c232583/app/code/Magento/Cms/Model/ResourceModel/Page.php#L200)

4. If you will provide array it will be just 1 and not merged with provided array. 

### Expected result (*)

1. If description correct and **store_id** must be array, then in Resource Model must be changed logic for merging array
2. If it must be just **int**, then it must be changed description for magic function in Magento\Cms\Model\Page

### Actual result (*)

1. See steps

**Comment by GovindaSharma** [comment]
@Sharkozp thanks for reporting this issue.It's look like its description should be updated because setStoreId is referring to single store ,if it is for multiple stores then method should be updates like setStoreIds .So accordingly description should be updated.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @GovindaSharma
Thank you for verifying the issue! :+1: Your confirmation will help us to acknowledge and process this report.

**Comment by Sharkozp** [comment]
@GovindaSharma and @magento-engcom-team ,

Yes, I also think so. For array it must be method **setStoreIds**, but I think it was added array for getting more than 1 page, because in **_getLoadSelect** store ids added via join and used for filters:
`->where('cps.store_id IN (?)', $store);`

[Link](https://github.com/magento/magento2/blob/bd82ed4ecd5c7d66c819711a2a4425d46c232583/app/code/Magento/Cms/Model/ResourceModel/Page.php#L237)

**Comment by mahesh-rajawat** [comment]
i am working on this.

**Comment by mahesh-rajawat** [comment]
@Sharkozp , as i think it must be type int, because `load()` function will always return a single record
matching with default store id and given store id.

**Comment by magento-engcom-team** [comment]
Hi @Sharkozp. Thank you for your report. 
The issue has been fixed in magento/magento2#22772 by @maheshWebkul721 in 2.3-develop branch
Related commit(s):
 - [ef5a50ec4b75cd40b9ec475c156fc4aaae266479](https://github.com/magento/magento2/commit/ef5a50ec4b75cd40b9ec475c156fc4aaae266479)
 - [876d6327b4bd17592cb2d4fb84e42bd625803f69](https://github.com/magento/magento2/commit/876d6327b4bd17592cb2d4fb84e42bd625803f69)

The fix will be available with the upcoming 2.3.3 release.

**Comment by magento-engcom-team** [comment]
Hi @Sharkozp. Thank you for your report. 
The issue has been fixed in magento/magento2#23149 by @krishprakash in 2.2-develop branch
Related commit(s):
 - [70d0992d84a9fe149550ca2a22de54e0f2e7b068](https://github.com/magento/magento2/commit/70d0992d84a9fe149550ca2a22de54e0f2e7b068)
 - [70aacf1090f7b8269f71a6be883bbfab2ba9d262](https://github.com/magento/magento2/commit/70aacf1090f7b8269f71a6be883bbfab2ba9d262)

The fix will be available with the upcoming 2.2.10 release.
